src/meta/credmeta.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2019 Elastos Foundation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+
+#include "ela_did.h"
+#include "did.h"
+#include "common.h"
+#include "credmeta.h"
+#include "JsonGenerator.h"
+#include "diddocument.h"
+#include "diderror.h"
+
+static const char *ALIAS = "DX-alias";
+
+int CredentialMetaData_ToJson_Internal(CredentialMetaData *metadata, JsonGenerator *gen)
+{
+    assert(metadata);
+    assert(gen);
+
+    return MetaData_ToJson_Internal(&metadata->base, gen);
+}
+
+const char *CredentialMetaData_ToJson(CredentialMetaData *metadata)
+{
+    assert(metadata);
+
+    return MetaData_ToJson(&metadata->base);
+}
+
+int CredentialMetaData_FromJson_Internal(CredentialMetaData *metadata, cJSON *json)
+{
+    assert(metadata);
+    assert(json);
+
+    return MetaData_FromJson_Internal(&metadata->base, json);
+}
+
+int CredentialMetaData_FromJson(CredentialMetaData *metadata, const char *data)
+{
+    assert(metadata);
+    assert(data);
+
+    return MetaData_FromJson(&metadata->base, data);
+}
+
+void CredentialMetaData_Free(CredentialMetaData *metadata)
+{
+    assert(metadata);
+
+    MetaData_Free(&metadata->base);
+}
+
+int CredentialMetaData_SetAlias(CredentialMetaData *metadata, const char *alias)
+{
+    assert(metadata);
```

Comment:
Please use "if" statement to check parameter in the export APIs instead of "assert".